### PR TITLE
Revert [Clang] prevent errors for deduction guides using deduced type aliases

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Reverts https://github.com/llvm/llvm-project/pull/117450

---

Revert https://github.com/llvm/llvm-project/pull/117450 because the prior behavior aligns with the standard - _`template-name`_ in the _`simple-template-id`_ in the return type shall be the actual name of a class template rather than some equivalent alias template. Details https://github.com/llvm/llvm-project/issues/54909#issuecomment-2508418355.

### CODE REVIEW

**Review by hokein — COMMENTED** [review]
The change looks good to me (please wait for @erichkeane  review before landing). I’d suggest adding a brief rationale in the commit message instead of just including links, to make it more self-contained.

**Review by cor3ntin — APPROVED** [review]

**Comment by a-tarasyuk** [comment]
@hokein thanks. I've updated the PR description.

**Review by hokein — APPROVED** [review]

**Review by erichkeane — APPROVED** [review]
Thank you very much for working so quickly on this!  LGTM.

**Pull request merged**
